### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Y-Maxym**

### CODE REVIEW

File: src/test/java/calculator/app/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package calculator.app;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (yura-jt) [inline src/test/java/calculator/app/MainTest.java]: wild cards import should not be used

File: src/test/java/calculator/app/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        assertEquals(expectedResult, obtainedResult);
+    }
+
+    @Test
+    @DisplayName("Test subtraction functionality")
+    void givenTwoNumbers_whenSubtractionFunctionality_thenCorrectResult() {
+        //given
+        Integer a = 10;
+        Integer b = 5;
+        Integer expectedResult = 5;
```
Comment (yura-jt) [inline src/test/java/calculator/app/MainTest.java]: please separate logic blocks:        
```
       //given
        Integer a = 10;
        Integer b = 5;
        Integer expectedResult = 50;
        
        //when
        Integer obtainedResult = main.multiplication(a, b);
        
        //then
        assertEquals(expectedResult, obtainedResult);
```

File: src/test/java/calculator/app/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        main = new Main();
+    }
+
+    @Test
+    @DisplayName("Test addition functionality")
+    void givenTwoNumbers_whenAdditionFunctionality_thenCorrectResult() {
+        //given
+        Integer a = 10;
+        Integer b = 5;
+        Integer expectedResult = 15;
```
Comment (yura-jt) [inline src/test/java/calculator/app/MainTest.java]: expected / actual is often used in tests.

Integer expected = 15; 

//when
Integer actual = main.addition(a, b);

File: src/test/java/calculator/app/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        assertEquals(expectedResult, obtainedResult);
+    }
+
+    @Test
+    @DisplayName("Test division by zero functionality")
+    void givenZero_whenDivisionFunctionality_thenExceptionThrown() {
+        //given
+        Integer a = 10;
+        Integer b = 0;
+        //when
```
Comment (yura-jt) [inline src/test/java/calculator/app/MainTest.java]: //when
ArithmeticException exception = assertThrows(ArithmeticException.class, () -> main.division(a, b));

//then
assertEquals(exception.getMessage(), "Division by zero");

File: src/main/java/calculator/app/Main.java [hunk 5]
```diff
@@ -1,4 +1,21 @@
 package calculator.app;
 
 public class Main {
+
```
Comment (yura-jt) [inline src/main/java/calculator/app/Main.java]: Main is not good place to put this code.
Imagine I'm your colleague and said "Hey, do you need Main functionality for your task?"

You will reply to me "What? What main functionality, what you are talking about".
How we can understand from class name - what class do?

Main is class that is entry point for execution program.

So please name class to Calculator, ArithmeticCalculator, BasicCalculator etc, but class name should be always correspond with its content

File: src/main/java/calculator/app/Main.java [hunk 6]
```diff
@@ -1,4 +1,21 @@
...
+    public Integer subtraction(Integer a, Integer b) {
+        return a - b;
+    }
+
+    public Integer multiplication(Integer a, Integer b) {
+        return a * b;
+    }
+
+    public Integer division(Integer a, Integer b) {
+        if (b == 0) throw new ArithmeticException("Division by zero");
```
Comment (yura-jt) [inline src/main/java/calculator/app/Main.java]: curly braces for IF
If .. {
  throw new ..
  }

File: src/main/java/calculator/app/Main.java [hunk 7]
```diff
@@ -1,4 +1,21 @@
...
+
+    public Integer subtraction(Integer a, Integer b) {
+        return a - b;
+    }
+
+    public Integer multiplication(Integer a, Integer b) {
+        return a * b;
+    }
+
+    public Integer division(Integer a, Integer b) {
```
Comment (yura-jt) [inline src/main/java/calculator/app/Main.java]: return type I guess could be double, to not loose precision of result

**Review by yura-jt — CHANGES_REQUESTED** [review]

File: src/test/java/calculator/app/CalculatorTest.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class CalculatorTest {
+
+    private static Calculator calculator;
+
+    @BeforeAll
+    static void setMain() {
```
Comment (yura-jt) [inline src/test/java/calculator/app/CalculatorTest.java]: void init() or void setup()

File: src/main/java/calculator/app/Calculator.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    public Integer multiplication(Integer a, Integer b) {
+        return a * b;
+    }
+
+    public Double division(Integer a, Integer b) {
+        if (b == 0) {
+            throw new ArithmeticException("Division by zero");
+        }
```
Comment (yura-jt) [inline src/main/java/calculator/app/Calculator.java]: don't forget to divide by empty line logic blocks.

Check for Divide by Zero and Exception it's one thing
Return and divide other.

File: src/test/java/calculator/app/CalculatorTest.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class CalculatorTest {
+
+    private static Calculator calculator;
```
Comment (yura-jt) [inline src/test/java/calculator/app/CalculatorTest.java]: there is no need to use static modifier in this case

File: src/main/java/calculator/app/Calculator.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
+package calculator.app;
```
Comment (yura-jt) [inline src/main/java/calculator/app/Calculator.java]: package com.calculator.app

com is very often used as root package

File: src/main/java/calculator/app/Calculator.java [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+public class Calculator {
+
+    public Integer addition(Integer a, Integer b) {
```
Comment (yura-jt) [inline src/main/java/calculator/app/Calculator.java]: Integer wrapper has two concerns:
1. Question arise why you don't use primitive int
2. Integer could be null, and your method should deal with it

I suggest just use int instead

File: src/test/java/calculator/app/CalculatorTest.java [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+    @Test
+    @DisplayName("Test subtraction functionality")
+    void givenTwoNumbers_whenSubtractionFunctionality_thenCorrectResult() {
+        //given
+        Integer a = 10;
+        Integer b = 5;
+        Integer expected = 5;
+
+        //when
+        Integer actual = calculator.subtraction(a, b);
```
Comment (yura-jt) [inline src/test/java/calculator/app/CalculatorTest.java]: one test for 1 method is not enough in this case, because we need to test differenet scenarios.

Always keep in mind, that unit tests and testing at all, should not be performed to prove that your program is working.

You should write tests like you want to find, that your code is NOT working.
In other words, imagine, someone propose you 100 000 dollars and say "Just find scenario, that will make your code fail".

It's different mindset, otherwise tests will be just bad and for checkbox.

In this case, you could use corner / edge case.
2 + 5 work, ok, what about 5 + 0? Zero is special number

Then what about negative numbers, -100 + -40 or -100 + 40

Then what about large number - 34930294 + 32538394

In business logic usually it's enough to cover conditions (IF ELSE branching) or HappyPath / sunnyday scenario when all is ok, and negative if it could be exception.

In this case we deal with numbers so we need to think about different possible scenarios.

To make easier and faster, I recommend use Parameterize testing, just use CsvSource or MethodSource and 3-4 scenario for each method (default, with zero, negative, large numbers)

File: src/main/java/com/calculator/app/Calculator.java [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+    public int multiplication(int a, int b) {
+        return a * b;
+    }
+
+    public double division(int a, int b) {
+        if (b == 0) {
+            throw new ArithmeticException("Division by zero");
+        }
+        return 1.0 * a / b;
```
Comment (yura-jt) [inline src/main/java/com/calculator/app/Calculator.java:22]:
```
        if (b == 0) {
            throw new ArithmeticException("Division by zero");
        }

        return 1.0 * a / b;
```
Try to structure your code in methods.

Like you divide functionality to 5 small methods, not 1 large.
In method you also divide code into logic blocks.

       if (b == 0) {
            throw new ArithmeticException("Division by zero");
        }
        That's one logic block, we validating input and throw exception
        
        return 1.0 ...
         it's other block, we perform divide and return.
         
          So let's just show it, by separating code blocks by empty line. 
           It's more easier to read, if I want to read about return, why I should read about exception validation and find when this block is ends.
           
            If I could see it immidietly.
            
             1 class is like book page.
              Just by empty line mark different paragraph, it's easier to read and focus on what you need.

**Review by yura-jt — CHANGES_REQUESTED** [review]
